src/components/rating/rating.ts

```diff
@@ -0,0 +1,322 @@
+import { html, LitElement, nothing } from 'lit';
+import {
+  property,
+  query,
+  queryAssignedElements,
+  state,
+} from 'lit/decorators.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { watch } from '../common/decorators/watch';
+import { Constructor } from '../common/mixins/constructor';
+import { EventEmitterMixin } from '../common/mixins/event-emitter';
+import { SizableMixin } from '../common/mixins/sizable';
+import { styles } from './rating.base.css';
+import { styles as bootstrap } from './rating.bootstrap.css';
+import { styles as fluent } from './rating.fluent.css';
+import { styles as indigo } from './rating.indigo.css';
+import { clamp } from '../common/util';
+import type IgcRatingSymbolComponent from './rating-symbol';
+import { themes } from '../../theming';
+
+export interface IgcRatingEventMap {
+  igcChange: CustomEvent<number>;
+  igcHover: CustomEvent<number>;
+}
+
+/**
+ * Rating provides insight regarding others' opinions and experiences,
+ * and can allow the user to submit a rating of their own
+ *
+ * @element igc-rating
+ *
+ * @fires igcChange - Emitted when the value of the control changes.
+ * @fires igcHover - Emitted when hover is enabled and the user mouses over a symbol of the rating.
+ *
+ * @csspart base - The main wrapper which holds all of the rating elements.
```

Comment:
We should add the other exposed parts here.

Reply:
@DiyanDimitrov I've added the rest of the parts.